src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -4292,6 +4292,112 @@ static QVariant fcnStringToArray( const QVariantList &values, const QgsExpressio
   return array;
 }
 
+static QVariant fcnJsonToMap( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QString str = QgsExpressionUtils::getStringValue( values.at( 0 ), parent );
+  QJsonDocument document = QJsonDocument::fromJson( str.toUtf8() );
+  if ( document.isNull() || !document.isObject() )
+    return QVariantMap();
+
+  return document.object().toVariantMap();
+}
+
+static QVariant fcnMapToJson( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QVariantMap map = QgsExpressionUtils::getMapValue( values.at( 0 ), parent );
+  QJsonObject object = QJsonObject::fromVariantMap( map );
+  QJsonDocument document( object );
+  return document.toJson( QJsonDocument::Compact );
+}
+
+static QVariant fcnHstoreToMap( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QString str = QgsExpressionUtils::getStringValue( values.at( 0 ), parent );
+  if ( str.isEmpty() )
+    return QVariantMap();
+  str = str.trimmed();
+
+  QVariantMap map;
+  QList<QString> bits;
+  QList<QString> seps;
+  seps << "=>" << ",";
```

Comment:
Do we have code in the postgres hstore handling which we can reuse here?

Reply:
The code here is derived from the postgres provider code to handle hstore, except it's been condensed to one function, instead of three, and slightly improved (for e.g., I think malformed hstore would freeze QGIS due to a logic issue in the postgres provider's hstore handling.

Reply:
Could we move this to a "hstore utils" class then, and fix the postgres version?